### ISSUE TITLE
IConsumer write method docs should say "non-blocking"

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4853]
|<img alt="detly's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| detly reported|
|-|-|
|Trac ID|trac#4853|
|Type|enhancement|
|Created|2011-02-07 01:30:08Z|

The t.i.i.IConsumer(Interface) API docs don't mention that "write" is non-blocking (like it is for ITransport). If it is, they should.

Attachments:

* [non-blocking-consumer-write.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/09d/09d0058cce33404ac3b927d204d403afeb2d5f3f/69117000e54364fde62922691d93deba91cece18.patch) (969 bytes) - added by exarkun on 2011-03-01 16:32:00Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__4853 4853
type__enhancement enhancement
reporter__detly detly
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1297042208000000 1297042208000000
changetime__1299634979000000 1299634979000000
version__None None
owner__exarkun exarkun
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, commit with NEWS file (.misc, I guess.)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [30896]) Add explicit documentation for IConsumer.write about the non-blockingness of that method

Author: exarkun
Reviewer: itamar
Fixes: [#4853](https://github.com/twisted/twisted/issues/4853)
